### PR TITLE
IPv4 and IPv6 domain docs.

### PR DESCRIPTION
**Pull request by Enmk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Other
Docs

Short description (up to few sentences):
IPv4 and IPv6 domains docs.
...

### CODE REVIEW

File: docs/en/data_types/domains/domains.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Domains
+
+Domains are special-purpose types, that add some extra features atop of existing 'base' type, leaving on-wire and on-disc format of underlying table intact. At the moment, Clickhouse does not support user-defined domains, and builtin domains are limited in functionality.
```
Comment (blinkov) [inline docs/en/data_types/domains/domains.md]: * Click**H**ouse
* I'd remove "and builtin domains are limited in functionality"
* Why single quotes around base?

File: docs/en/data_types/domains/domains.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+You can use domains in anywhere corresponding base type can be used:
```
Comment (blinkov) [inline docs/en/data_types/domains/domains.md]: "in anywhere" sounds incorrect, maybe "in any place where"?

File: docs/en/data_types/domains/domains.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+* Use it as index if base type can be used as index
+* Call functions with values of domain column
+* etc.
+
+### Extra features of domains
+
+* Explicit column type name in `SHOW CREATE TABLE` or `DESCRIBE TABLE`
+* Custom input format for `INSERT INTO domain_table(domain_column) VALUES(...)`
+* Custom output format for `SELECT domain_column FROM domain_table`
+* Loading data from external source: `INSERT INTO domain_table FORMAT CSV ...`
```
Comment (blinkov) [inline docs/en/data_types/domains/domains.md]: Actually these are called input/output formats in the rest of documentation, not the features from previous two bullets. Those are more like "Input from human-readable representation for INSERT INTO..." and "Output to human-readable representation for SELECT...".

File: docs/en/data_types/domains/IPv4.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
+## IPv4
+`IPv4` is a domain based on `UInt32` type, this is basically a typed replacement of `UInt32` for storing IPv4 values. It provides compact storage (compared to `String`), human-friendly output (compared to `UInt32`) and column type information on inspection.
+
+### Basic usage
```
Comment (blinkov) [inline docs/en/data_types/domains/IPv4.md]: Title case for all English headers: Basic **U**sage
Here and in many places below

File: docs/en/query_language/functions/ip_address_functions.md [hunk 5]
```diff
@@ -147,6 +147,70 @@ SELECT
 └─────────────────────────────────────┴─────────────────────┘
 ```
 
+## toIPv4(string)
+
+An alias to `IPv4StringToNum()` that takes a string form of IPv4 address and returns value of [`IPv4`](../../data_types/domains/IPv4.md) type, which is binary equal to value returned by `IPv4StringToNum()`.
```
Comment (blinkov) [inline docs/en/query_language/functions/ip_address_functions.md]: remove backticks around IPv4

File: docs/en/data_types/domains/domains.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Domains
```
Comment (blinkov) [inline docs/en/data_types/domains/domains.md]: * New files should be added into `toc_en.yml`
* Not-yet-translated languages should have similar files as relative symlinks to English documents and also added to `toc_*.yml`
* domains/domains.md will have weird URL with double "domains", so it'd be better to rename to overview.md or something like that

**Review by blinkov — CHANGES_REQUESTED** [review]

**Comment by blinkov** [comment]
Also there are new errors in links integrity check:

> INFO:root:Link to nowhere: #../../data_types/domains/IPv4/
> INFO:root:Link to nowhere: #../../data_types/domains/IPv6/

(probably because files are named in lowercase)

File: docs/en/data_types/domains/IPv4.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+## IPv4
+`IPv4` is a domain based on `UInt32` type, this is basically a typed replacement of `UInt32` for storing IPv4 values. It provides compact storage (compared to `String`), human-friendly output (compared to `UInt32`) and column type information on inspection.
```
Comment (BayoNet) [inline docs/en/data_types/domains/IPv4.md]: IPv4 doesn't look as `SQL domain`. It looks like usual embedded data type, while domain is a custom entity, that user define within their database with `CREATE DOMAIN` clause.

Comment (Enmk) [reply docs/en/data_types/domains/IPv4.md]: Yep, I've mention that in the `overview.md`, `IPv4` and `IPv6` are kind of builtin domains, while there is still much work to do in order to support full-fledged user-defined domains.

File: docs/en/data_types/domains/overview.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+You can use domains anywhere corresponding base type can be used:
+
+* Create a column of domain type
+* Read/write values from/to domain column
+* Use it as index if base type can be used as index
+* Call functions with values of domain column
+* etc.
+
+### Extra features of domains
```
Comment (blinkov) [inline docs/en/data_types/domains/overview.md]: Extra **F**eatures of **D**omains

**Review by blinkov — CHANGES_REQUESTED** [review]
+ it looks like some previous comments are not addressed yet

File: docs/en/data_types/domains/IPv4.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+```
+┌─toTypeName(from)─┬─hex(from)─┐
+│ IPv4             │ B7F7E83A  │
+└──────────────────┴───────────┘
+```
+
+
+If you want to convert IPv4 value to a string, you have to do that explicitly with `IPv4NumToString()` function:
```
[the remark this answers is not in the document]
Comment (Enmk) [reply docs/en/data_types/domains/IPv4.md]: Fixed

File: docs/en/data_types/domains/IPv4.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+```
+┌─s──────────────┬─toTypeName(IPv4NumToString(from))─┐
+│ 183.247.232.58 │ String                            │
+└────────────────┴───────────────────────────────────┘
+```
+
+### Functions
+Any function that takes `UInt32` arguments can also take `IPv4` arguments too
```
[the remark this answers is not in the document]
Comment (Enmk) [reply docs/en/data_types/domains/IPv4.md]: Fixed that by removing the whole section.

File: docs/en/data_types/domains/IPv4.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+Or even math operations
+``` sql
+SELECT url, from, CAST(from + 1 AS IPv4) FROM hits LIMIT 1;
+
+┌─url───────────────────────┬───────────from─┬─CAST(plus(from, 1), 'IPv4')─┐
+│ https://clickhouse.yandex │ 183.247.232.58 │              183.247.232.59 │
+└───────────────────────────┴────────────────┴─────────────────────────────┘
+```
+
+!!! warning BUG
```
[the remark this answers is not in the document]
Comment (Enmk) [reply docs/en/data_types/domains/IPv4.md]: Done

File: docs/en/data_types/domains/IPv6.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
+## IPv6
+`IPv6` is a domain based on `FixedString(16)` type, this is basically a typed replacement of `FixedString(16)` for storing IPv6 values. It provides compact storage (compared to `String`), human-friendly output (compared to `FixedString(16)`) and column type information on inspection.
```
[the remark this answers is not in the document]
Comment (Enmk) [reply docs/en/data_types/domains/IPv6.md]: Done

File: docs/en/data_types/domains/IPv6.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+└───────────────────────────────────────────────────────────────┴─────────────────────┘
+```
+
+!!! warning "Performance"
+    Please note that column values might be changed on disk, and hence this operation might be really I/O intensive.
+
+
+### Loading data from external source
+
+You can make Clikchouse load `IPv6` values from external source with `INSERT INTO ... FORMAT`:
```
[the remark this answers is not in the document]
Comment (Enmk) [reply docs/en/data_types/domains/IPv6.md]: Removed that section

File: docs/en/data_types/domains/domains.md (hunk 3 above)
Comment (Enmk) [reply docs/en/data_types/domains/domains.md]: Fixed

File: docs/en/data_types/domains/IPv4.md (hunk 4 above)
Comment (Enmk) [reply docs/en/data_types/domains/IPv4.md]: Done

File: docs/en/query_language/functions/ip_address_functions.md (hunk 5 above)
Comment (Enmk) [reply docs/en/query_language/functions/ip_address_functions.md]: Done

File: docs/en/data_types/domains/domains.md (hunk 6 above)
Comment (Enmk) [reply docs/en/data_types/domains/domains.md]: Fixed

**Comment by Enmk** [comment]
* Simplified documentation to meet style of other docs - removed superfluous sections.
* Added doc files to toc_{ru,en,zh,fa}.yml
* Added docs in other languages as symbolic links to English docs. 
* Added a script that helps in adding links.

File: docs/en/data_types/domains/overview.md (hunk 8 above)
Comment (Enmk) [reply docs/en/data_types/domains/overview.md]: Done

**Pull request merged**